tests/regression/lp-1828354/task.yaml

```diff
@@ -0,0 +1,206 @@
+summary: mount event propagation works inside snapd-created tmpfs
+# NOTE: The limitation comes from availability of Python 3.7 for mountinfo-tool
+systems: [ubuntu-18.04-64, ubuntu-core-18-64]
+details: |
+    When snapd mounts a tmpfs the sharing on the tmpfs is incorrect and breaks
+    per-user mount namespaces since events don't propagate there correctly.
+environment:
+    VARIANT/mimic: mimic
+    VARIANT/layout: layout
+    snap/mimic: test-snapd-app-mimic
+    snap/layout: test-snapd-app-layout
+prepare: |
+    # Enable persistence of per-user-mount-namespaces.
+    #
+    # This allows us to see persistent state of the per-user mount namespace.
+    # This test relies on this because otherwise per-user mount namespace is always
+    # re-created from per-snap mount namespace, on each command invocation.
+    snap set system experimental.per-user-mount-namespace=true
+
+    # Install our primary snap and connect it to mount-observe.
+    # We need the interface connection for reading mountinfo table.
+    snap pack "$snap"
+    snap install --dangerous "./${snap}_1_all.snap"
+    snap connect "$snap:mount-observe"
+
+    # Install the supporting content snap.
+    snap pack test-snapd-content
+    snap install --dangerous ./test-snapd-content_1_all.snap
+restore: |
+    # TODO: when snap unset is supported unset instead of explicitly disabling.
+    snap set system experimental.per-user-mount-namespace=false
+
+    rm -f "${snap}_1_all.snap"
+    snap remove "$snap"
+
+    rm -f test-snapd-content_1_all.snap
+    snap remove test-snapd-content
+
+    # Remove the private temporary directories we created.
+    rm -rf "/tmp/snap.$snap"
+execute: |
+    #shellcheck source=tests/lib/dirs.sh
+    . "$TESTSLIB/dirs.sh"
+
+    # Define helpers that run a command inside the snap as root or test user.
+    # The call to "su root -c" is redundant but makes for nice parity with the
+    # second function below.
+    as_snap_root() {
+        su root -c "snap run \"$snap.sh\" -c \"$*\""
+    }
+
+    as_snap_user() {
+        su test -c "snap run \"$snap.sh\" -c \"$*\""
+    }
+
+    # Depending on the test variant define different assert functions.
+    case "$VARIANT" in
+        mimic)
+            # Define properties that should hold in at all times.
+            assert_invariant() {
+                # Existing canary files are not clobbered.
+                as_snap_root /usr/bin/test -f "/snap/$snap/x1/canary"
+                as_snap_root /usr/bin/test -f "/snap/$snap/x1/dir/canary"
+                as_snap_root /usr/bin/test -f "/snap/$snap/x1/meta/canary"
+                test "$(as_snap_root /bin/cat "/snap/$snap/x1/canary")" = "app:canary"
+                test "$(as_snap_root /bin/cat "/snap/$snap/x1/dir/canary")" = "app:dir/canary"
+                test "$(as_snap_root /bin/cat "/snap/$snap/x1/meta/canary")" = "app:meta/canary"
+
+                # Same as above but for snap user in per-user mount namespace.
```

Comment:
Since all read only areas those relate to the writable mimic where sharing was broken. I agree it is worth adding tests that show how /run or /usr behaves.